tests/Network/Wai/Middleware/OpenApi/PathMapSpec.hs

```diff
@@ -0,0 +1,174 @@
+module Network.Wai.Middleware.OpenApi.PathMapSpec
+  ( spec
+  ) where
+
+import Prelude
+
+import Control.Monad (guard)
+import Data.ByteString (ByteString)
+import Data.ByteString.Char8 qualified as BS8
+import Data.Foldable (toList)
+import Data.Maybe (isNothing)
+import Data.String (IsString)
+import Network.Wai.Middleware.OpenApi.PathMap qualified as PathMap
+import Test.Hspec
+import Test.QuickCheck
+
+spec :: Spec
+spec =
+  describe "PathMap" $ do
+    it "empty matches nothing" $
+      property $
+        \path ->
+          isNothing $ PathMap.lookup @Bool (renderPath path) PathMap.empty
+
+    it "case 1" $
+      property $
+        \segment ->
+          PathMap.lookup
+            (renderPath Path {segments = [segment]})
+            (PathMap.fromList [("/{}", True)])
+            == Just True
+
+    it "case 2" $
+      property $
+        forAll (arbitrary `suchThat` (/= "a")) $ \segment ->
+          let result =
+                PathMap.lookup
+                  (renderPath $ Path [segment])
+                  (PathMap.fromList [("/a", False), ("/{}", True)])
+          in  counterexample ("result = " <> show result) (result == Just True)
+
+    it "case 3" $
+      property $
+        forAll (arbitrary `suchThat` (/= "a")) $ \segment ->
+          let result =
+                PathMap.lookup
+                  (renderPath $ Path [segment])
+                  (PathMap.fromList [("/{}", False), ("/a", True)])
+          in  counterexample ("result = " <> show result) (isNothing result)
```

Comment:
The funky case as described in https://github.com/freckle/wai-middleware-openapi/pull/1#discussion_r1800584356 - the spec written here describes what the code presently does, not what it should do.